core/src/main/java/com/amplifyframework/core/Amplify.java

```diff
@@ -179,6 +170,20 @@ private static void beginInitialization(@NonNull Category<? extends Plugin<?>> c
         updatePluginRegistry(plugin, RegistryUpdateType.REMOVE);
     }
 
+    /**
+     * Starts listening for a shake event and shows the developer menu on device shake.
+     * @param context an Android Context
+     */
+    public static void enableDeveloperMenu(Context context) {
+        // Start activity for developer menu on device shake
+        Intent mainIntent = new Intent(context, DeveloperMenuActivity.class);
+        mainIntent.setAction(Intent.ACTION_MAIN);
+        mainIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        ShakeDetector shakeDetector = new ShakeDetector(context, () -> context.startActivity(mainIntent),
+                true);
```

Comment:
Just an idea -- feel free to ignore --

Instead of using the boolean, you could encode this behavior in your callback:
```java
final class ShakeDetector {
    private final ShakeDetector.Listener listener;

    ShakeDetector(ShakeDetector.Listener listener) {
        this.listener = listener;
    }

    // Change this
    @FunctionalInterface
    interface Listener {
        void onShakeDetected(ShakeDetector source);
    }
}
```

Then you'll have an instance to call `stopDetecting()` on, in your lambda:
```java
ShakeDetector detector = new ShakeDetector(context, it -> {
    context.startActivity(mainIntent);
    it.stopDetecting();
});
```

Without changing the listener, you could already do:
```java
final AtomicReference<ShakeDetector.Listener> listener = new AtomicReference<>();
ShakeDetector detector = new ShakeDetector(context, () -> {
    context.startActivity(mainIntent);
    listener.get().stopDetecting();
});
```

^ This would give you a `final` reference inside of your lambda. The `AtomicReference` thing is sort of a kludge that I try to avoid if I can, though.

Reply:
Thanks for the suggestion. Using the FunctionalInterface works well and it does seem cleaner than using a boolean.